test/suite/hook_tiered.py

```diff
@@ -0,0 +1,145 @@
+# cause a test error.
+def tty(s):
+    WiredTigerTestCase.tty(s)
+
+# These are the hook functions that are run when particular APIs are called.
+
+# Called to replace Session.create
+def session_create_replace(arg, orig_session_create, session_self, uri, config):
+    if config == None:
+        base_config = ""
+    else:
+        base_config = config
+
+    # If the test is creating a table (not colstore or lsm), create a tiered table instead,
+    # using arg to determine number of tiers. Otherwise just do the create as normal.
+    #
+    # NOTE: the API for creating a tiered table will change in WT-7173 and this will need to
+    # be updated.
+    if uri[:6] == "table:" and "key_format=r" not in base_config and "type=lsm" not in base_config:
+        tier_string = ""
+        for i in range(arg):
+            new_uri = uri.replace('table:', 'file:tier' + str(i) + '_')
+            ret = orig_session_create(session_self, new_uri, config)
+            if ret != 0:
+                return ret
+            tier_string = tier_string + '"' + new_uri + '", '
+        tier_config = 'type=tiered,tiered=(tiers=(' + tier_string[0:-2] + ')),' + base_config
+        WiredTigerTestCase.verbose(None, 3,
+            'Creating tiered table {} with config = \'{}\''.format(uri, tier_config))
+        ret = orig_session_create(session_self, uri, tier_config)
+    else:
+        ret = orig_session_create(session_self, uri, config)
+    return ret
+
+# Called to replace Session.drop
+def session_drop_replace(arg, orig_session_drop, session_self, uri, config):
+    # Drop isn't implemented for tiered tables.  Only do the delete if this could be a
+    # uri we created a tiered table for.  Note this isn't a precise when we did/didn't
+    # create a tiered table, but we don't have the create config around to check.
+    ret = 0
+    if (uri[:6] != 'table:'):
+        ret = orig_session_drop(session_self, uri, config)
+    return ret
+
+# Called to replace Session.verify
+def session_verify_replace(arg, orig_session_verify, session_self, uri):
+    return 0
+
+# Every hook file must have one or more classes descended from WiredTigerHook
+# This is where the hook functions are 'hooked' to API methods.
+class TieredHookCreator(wthooks.WiredTigerHookCreator):
+    def __init__(self, arg=0):
+        # Argument specifies the number of tiers to test. The default is 2.
+        if arg == None:
+            self.arg = 2
+        else:
+            self.arg = int(arg)
+
+    # Is this test one we should skip? We skip tests of features supported on standard
+    # tables but not tiered tables, specififically cursor caching and checkpoint cursors.
+    def skip_test(self, test):
+        skip = ["test_cursor13_reopens",
+                "test_cursor13_big",
+                "test_cursor13_drops",
+                "test_cursor13_dup",
+                "checkpoint"]
+        match = [n for n in skip if n in str(test)]
+        return match
```

Comment:
I don't think you need to build a `match` list here, so list comprehension is probably a distraction.
I think I'd go with:
```
   for n in skip:
     if n in str(test):
       return False
   return True
```
It's probably easier for non-Python-experts to understand.

Reply:
Done.  List comprehension is one of the fun hammers in Python. So I'm sometimes too eager to find nails...